subsys/net/lib/tftp/tftp_client.c

```diff
@@ -0,0 +1,306 @@
+ * -> ACK_OPCODE:   If the remote server responded with "Ack".
+ */
+static int tftp_send_request(int sock, u8_t request,
+			     const char *remote_file, const char *mode)
+{
+	u8_t    retx_cnt = 0;
+	u32_t   req_size;
+
+	/* Create TFTP Request. */
+	req_size = make_request(remote_file, mode, request);
+
+send_req:
+
+	/* Send this request to the server. */
+	if (send(sock, tftpc_buffer, req_size, 0)) {
+		if (tftpc_recv_data(sock)) {
+			return sys_get_be16(tftpc_buffer);
+		}
+	}
+
+	/* No of times we had to re-tx this "request". */
+	retx_cnt++;
+
+	/* Log this out. */
+	LOG_DBG("Unable to get data from the TFTP Server.");
+	LOG_DBG("no_of_retransmists = %d", retx_cnt);
+
+	/* Are we retransmitting? */
+	if (retx_cnt < TFTP_REQ_RETX) {
+		LOG_DBG("Are we re-transmitting: YES");
+		goto send_req;
+	}
+
+	LOG_DBG("Are we re-transmitting: NO");
+	return ERROR_OPCODE;
+}
+
+/* Name: tftp_get
+ * Description: This function gets "file" from the remote server.
+ */
+int tftp_get(struct sockaddr *server, struct tftpc *client,
+	     const char *remote_file, const char *mode)
+{
+
+	s32_t   stat     = TFTPC_UNKNOWN_FAILURE;
+	u8_t    retx_cnt = 0;
+	s32_t   sock;
+
+	/* Re-init the global "block number" variable. */
+	tftpc_block_no = 1;
+	tftpc_index    = 0;
+
+	/* Create Socket for TFTP (IPv4 / UDP) */
+	sock = socket(server->sa_family, SOCK_DGRAM, IPPROTO_UDP);
+	if (sock < 0) {
+		LOG_ERR("Failed to create UDP socket (IPv4): %d", errno);
+		return -errno;
+	}
+
+	/* Connect with the address.  */
+	stat = connect(sock, server, sizeof(struct sockaddr_in6));
+	if (stat < 0) {
+		LOG_ERR("Cannot connect to UDP remote (IPv4): %d", errno);
+		return -errno;
+	}
+
+	/* Obtain Global Lock before accessing critical resources. */
+	k_mutex_lock(&tftpc_lock, K_FOREVER);
+
+	/* Send out the request to the TFTP Server. */
+	stat = tftp_send_request(sock, READ_REQUEST, remote_file, mode);
+	if (stat == ERROR_OPCODE) {
+		LOG_ERR("Server responded with error.");
+		return TFTPC_REMOTE_ERROR;
+	}
+
+process_resp:
+
+	/* Process server response. */
+	stat = tftpc_process_resp(sock, client);
+	if (stat == TFTPC_BUFFER_OVERFLOW ||
+	    stat == TFTPC_SUCCESS) {
+		goto req_done;
+	}
+
+tftpc_recv:
+
+	/* More data is available - Read (or we read a duplicate). */
+	stat = tftpc_recv_data(sock);
+	if (stat <= 0) {
+		/* No response from server. */
+		LOG_DBG("Server response timeout.");
+
+		/* Retries exhausted ? */
+		if (++retx_cnt >= TFTP_REQ_RETX) {
+			LOG_ERR("No more retransmits available. Exiting");
+			return TFTPC_RETRIES_EXHAUSTED;
+		}
+
+		/* Start Retransmission. */
+		LOG_DBG("Starting Re-transmission.");
+		send_ack(sock, tftpc_block_no);
+		goto tftpc_recv;
+	}
+
+	/* Received response from the server. */
+	LOG_DBG("Received data of size: %d", stat);
+	goto process_resp;
+
+req_done:
+	/* Clean up. */
+	k_mutex_unlock(&tftpc_lock);
+	close(sock);
+	return stat;
+}
+
+/* Name: tftp_get
```

Comment:
Please remove this dummy function (returning success status) and copy-pasted incorrect comment.